iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/bicep/PrimaryTypeExpressionTest.java

```diff
@@ -265,12 +291,40 @@ void shouldParseSimpleObjectType() {
 
   @Test
   void shouldParseSimpleTupleType() {
-    String code = code("[",
-      "@functionName123() typeExpr",
-      "]");
+    String code = """
+      [
+      @functionName123() typeExpr
+      ]""";
     TupleType tree = parse(code, BicepLexicalGrammar.TUPLE_TYPE);
     assertThat(tree.is(ArmTree.Kind.TUPLE_TYPE)).isTrue();
     Assertions.assertThat(recursiveTransformationOfTreeChildrenToStrings(tree))
       .containsExactly("[", "@", "functionName123", "(", ")", "typeExpr", "]");
   }
+
+  @Test
+  void shouldParseArrayType() {
+    String code = "string[3]";
+    ArrayTypeReference tree = parse(code, BicepLexicalGrammar.PRIMARY_TYPE_EXPRESSION);
+    assertThat(tree.value()).isEqualTo("string[3]");
+    assertThat(tree.is(ArmTree.Kind.ARRAY_TYPE_REFERENCE)).isTrue();
+    assertThat(recursiveTransformationOfTreeChildrenToStrings(tree)).containsExactly("string", "[", "3", "]");
+  }
+
+  @Test
+  void shouldParseArrayTypeWithoutLength() {
+    String code = "string[]";
+    ArrayTypeReference tree = parse(code, BicepLexicalGrammar.PRIMARY_TYPE_EXPRESSION);
+    assertThat(tree.value()).isEqualTo("string[]");
+    assertThat(tree.is(ArmTree.Kind.ARRAY_TYPE_REFERENCE)).isTrue();
+    assertThat(recursiveTransformationOfTreeChildrenToStrings(tree)).containsExactly("string", "[", "]");
+  }
+
+  @Test
+  void shouldParseMultiDimensionalArrayType() {
+    String code = "string[3][5][7]";
+    ArrayTypeReference tree = parse(code, BicepLexicalGrammar.PRIMARY_TYPE_EXPRESSION);
+    assertThat(tree.value()).isEqualTo("string[3][5][7]");
+    assertThat(tree.is(ArmTree.Kind.ARRAY_TYPE_REFERENCE)).isTrue();
+    assertThat(recursiveTransformationOfTreeChildrenToStrings(tree)).containsExactly("string", "[", "3", "]", "[", "5", "]", "[", "7", "]");
+  }
```

Comment:
I would have moved this test in the dedicated test class `ArrayTypeReferenceTest`.